astar.py

```diff
@@ -112,7 +115,8 @@ def getNeighbors(node, nodes):
 
   for s in spaces:
       if s[0] >= 0 and s[1] >= 0 and s[0] < len(data[0]) and s[1] < len(data):

```

Comment:
This conditional isn't immediately readable and a comment may be warranted. Another option is to this up into e.g., two boolean variables with meaningful names. E.g, 

```
has_neighbors = s[0] >= 0 and s[1] >= 0
neighbors_fewer_than_data = s[0] < len(data[0]) and s[1] < len(data)
if has_neighbors and neighbors_fewer_than_data:
    ...
```

(use better variable names though :laughing: )